core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -1,143 +1,141 @@
+     */
+    fun getBestTilesToFoundCity(unit: MapUnit, distanceToSearch: Int? = null): Pair<HashMap<Tile, Float>, Pair<Tile?, Float>> {
+        val range =  if (distanceToSearch != null) distanceToSearch else {
+            val distanceFromHome = if (unit.civ.cities.isEmpty()) 0
+            else unit.civ.cities.minOf { it.getCenterTile().aerialDistanceTo(unit.getTile()) }
+            (8 - distanceFromHome).coerceIn(1, 5) // Restrict vision when far from home to avoid death marches
+        }
+        val nearbyCities = unit.civ.gameInfo.getCities()
+            .filter { it.getCenterTile().aerialDistanceTo(unit.getTile()) <= 7 + range }
 
         val possibleCityLocations = unit.getTile().getTilesInDistance(range)
-            .filter { canUseTileForRanking(it, unit.civ) }
-            .filter {
-                val tileOwner = it.getOwner()
-                it.isLand && !it.isImpassible() && (tileOwner == null || tileOwner == unit.civ) // don't allow settler to settle inside other civ's territory
-                    && (unit.currentTile == it || unit.movement.canMoveTo(it))
-                    && it !in tilesNearCities
+            .filter { canSettleTile(it, unit.civ, nearbyCities) && (unit.getTile() == it || unit.movement.canMoveTo(it)) }
+        val uniqueCache = LocalUniqueCache()
+        val baseTileMap = HashMap<Tile, Float>()
+        val tileRankMap = HashMap<Tile, Float>()
+        var maxValueTile: Tile? = null
+        var maxValueRank: Float = 0f
+        for (tile in possibleCityLocations) {
+            val tileValue = rankTileToSettle(tile, unit.civ, nearbyCities, baseTileMap, uniqueCache)
+            if (tileValue > maxValueRank) {
+                maxValueTile = tile
+                maxValueRank = tileValue
             }
+            tileRankMap[tile] = tileValue
+        }
+        return Pair(tileRankMap, Pair(maxValueTile, maxValueRank))
+    }
 
-        val luxuryResourcesInCivArea = getLuxuryResourcesInCivArea(unit.civ)
-
-        return possibleCityLocations
-            .map {
-                it to
-                    rankTileAsCityCenterWithCachedValues(
-                        it,
-                        nearbyTileRankings,
-                        luxuryResourcesInCivArea,
-                        unit.civ
-                    )
+    private fun canSettleTile(tile: Tile, civ: Civilization, nearbyCities: Sequence<City>): Boolean {
+        val modConstants = civ.gameInfo.ruleset.modOptions.constants
+        if (!tile.isLand || tile.isImpassible()) return false
+        if (tile.getOwner() != null && tile.getOwner() != civ) return false
+        if (nearbyCities.any {
+                it.getCenterTile().aerialDistanceTo(tile) <=
+                    if (tile.getContinent() == it.getCenterTile().getContinent()) modConstants.minimalCityDistance
+                    else modConstants.minimalCityDistanceOnDifferentContinents
+            }) return false
```

Comment:
Turn this into a for loop, please, then we can have the distance as a val and everything will be clearer

for city in nearby cities
distance = if () a else b
if distancetotile < distance return false